### PR TITLE
cli: use Cobra's positional argument validation

### PR DESCRIPTION
**Pull request by nvb**
Positional argument validation was added to spf13/cobra in
spf13/cobra#284, but our cli still
relies on custom validation in each of the "run" functions.
This validation is unstandardized and overly permissive in
certain cases. This change fixes this by replacing our custom
logic with validation expressed using Cobra's `PositionalArgs`
validators.

Release note: None

### CODE REVIEW

File: pkg/cli/cert.go [hunk 1]
```diff
@@ -84,6 +85,12 @@ Requires a CA cert in "<certs-dir>/ca.crt" and matching key in "--ca-key".
 If "ca.crt" contains more than one certificate, the first is used.
 Creation fails if the CA expiration time is before the desired certificate expiration.
 `,
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return errors.Errorf("no hosts specified. Need at least one")
```
Comment (knz) [inline pkg/cli/cert.go]: `create-node requires at least one host name or address, none was specified`

File: pkg/cli/debug.go [hunk 2]
```diff
@@ -55,11 +55,12 @@ import (
 )
 
 var debugKeysCmd = &cobra.Command{
-	Use:   "keys [directory]",
+	Use:   "keys <directory>",
```
Comment (knz) [inline pkg/cli/debug.go]: omg thank you thank you this has been bugging me for ages

**Review by knz — APPROVED** [review]
You'll need to remove the now-unused `usageAndError` to please the linter.

File: pkg/cli/cert.go (hunk 1 above)
Comment (nvanbenschoten) [reply pkg/cli/cert.go]: Done.

**Comment by nvanbenschoten** [comment]
> You'll need to remove the now-unused usageAndError to please the linter.

Done.

TFTR!

File: pkg/cli/debug.go (hunk 2 above)
Comment (nvanbenschoten) [reply pkg/cli/debug.go]: me too!

**Pull request merged**